src/journal/journald-server.c

```diff
@@ -1156,12 +1193,77 @@ void server_dispatch_message(
         dispatch_message_real(s, iovec, n, m, c, tv, priority, object_pid);
 }
 
+static int do_flush(Server *s, sd_journal *j) {
+        usec_t before_flush;
+        unsigned n = 0;
+        int r;
+
+        assert(s && s->system_journal);
```

Comment:
please make these two asserts.

we generally want one assert check per test. That makes debugging a lot easier if an assert is actually hit. because otherwise you won't know if `s` or `s->system_journal` ended up being NULL here.